src/App.jsx

```diff
@@ -14,33 +16,84 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPreparedGoods(goods, field, reversed) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (field) {
+    preparedGoods.sort((good1, good2) => {
+      switch (field) {
+        case 'alphabet':
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+        case 'length':
+          return good1.length - good2.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [field, setField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, field, reversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setField('alphabet')}
```

Comment:
suggest to make constants for sort fields and use everywhere.
It can be object with all possible fields.
```
const SORT_FIELD = {
   None: '',
   Alphabet: 'alphabet',
   Lenght: 'length'
}